bigtable/tests/unit/test_row_data.py

```diff
@@ -641,12 +659,28 @@ def _flatten_cells(prd):
     from google.cloud._helpers import _bytes_to_unicode
     from google.cloud._helpers import _microseconds_from_datetime
 
-    for row_key, row in prd.rows.items():
-        for family_name, family in row.cells.items():
+    try:
+        # Flatten PartialRowsData
+        for row_key, row in prd.rows.items():
+            for family_name, family in row.cells.items():
+                for qualifier, column in family.items():
+                    for cell in column:
+                        yield {
+                            u'rk': _bytes_to_unicode(row_key),
+                            u'fm': family_name,
+                            u'qual': _bytes_to_unicode(qualifier),
+                            u'ts': _microseconds_from_datetime(cell.timestamp),
+                            u'value': _bytes_to_unicode(cell.value),
+                            u'label': u' '.join(cell.labels),
+                            u'error': False,
+                        }
+    except AttributeError:
+        # Flatten PartialRowData
+        for family_name, family in prd.cells.items():
             for qualifier, column in family.items():
                 for cell in column:
                     yield {
-                        u'rk': _bytes_to_unicode(row_key),
+                        u'rk': _bytes_to_unicode(prd.row_key),
                         u'fm': family_name,
                         u'qual': _bytes_to_unicode(qualifier),
                         u'ts': _microseconds_from_datetime(cell.timestamp),
```

Comment:
Haven't thought of a good way to refactor this yet due to the yield statements...  Suggestions would be welcome.